### PR TITLE
Go Implementation of Heapsort

### LINKED ISSUE #2147: Heap Sort
- [x] C
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [x] Dart
- [x] Go - @Shr03mink
- [x] Java
- [x] JS
- [ ] Kotlin
- [x] Php - @anmolkaur18
- [x] Python
- [x] Readme
- [x] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Shr03mink**
Fixes #2147

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  Implemented heap sort by creating Max heap
-  Corrected Indentation

#### Languages Used:
-  Go
- 

#### Files Added:
-  heapsort.go
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Heap_Sort/heapsort.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/*    Heap Sort is a comparison-based sorting method performed on a Heap data structure. 
+      Here we build a max heap first to later sort the elements in the heap in ascending order.
+*/
+
```
Comment (singh-shreya6) [inline Heap_Sort/heapsort.go:4]: One blank line is enough

File: Heap_Sort/heapsort.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+   
+
+package main 
+import "fmt" 
+
+
+func max_heapify(arr []int, n int, i int) { 
+
+    largest := i // Initialize largest as root 
+    l := 2 * i + 1 // left child
```
Comment (singh-shreya6) [inline Heap_Sort/heapsort.go:13]: One blank line is enough

File: Heap_Sort/heapsort.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if largest != i { 
+        
+        //swapping
+        arr[i], arr[largest] = arr[largest], arr[i] 
+  
+        // Recursively heapify the affected sub-tree 
+        max_heapify(arr, n, largest) 
+    } 
+} 
+
```
Comment (singh-shreya6) [inline Heap_Sort/heapsort.go:36]: One blank line is enough

**Comment by Shr03mink** [comment]
I have squashed the commits. Could you please review and let me know if there's any mistake or any other changes are required.

**Comment by singh-shreya6** [comment]
I have merged your commit. Please update your name on your profile. Need to add in scoresheet

**Comment by Shr03mink** [comment]
Done.

**Pull request merged**